modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -111,7 +108,7 @@
  * @since 4.0
  */
 public abstract class ExtensionComponent<T extends ComponentModel> extends AbstractComponent
-    implements MuleContextAware, MetadataKeyProvider, MetadataProvider<T>, ComponentValueProvider,
+    implements MuleContextAware, MetadataKeyProvider, MetadataProvider<T>, SdkComponentValueProvider,
```

Comment:
This breaks backwards compatibility. What happens with current clients which are casting to `ComponentValueProvider`?  Where's the adaptation layer for that?